extension/encoding/awslogsencodingextension/internal/unmarshaler/elb-access-log/unmarshaler.go

```diff
@@ -0,0 +1,254 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package elbaccesslogs
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/otel/semconv/v1.27.0"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/internal/unmarshaler"
+)
+
+type elbAccessLogUnmarshaler struct {
+	buildInfo component.BuildInfo
+	logger    *zap.Logger
+}
+
+func NewELBAccessLogUnmarshaler(buildInfo component.BuildInfo, logger *zap.Logger) unmarshaler.AWSUnmarshaler {
+	return &elbAccessLogUnmarshaler{
+		buildInfo: buildInfo,
+		logger:    logger,
+	}
+}
+
+// UnmarshalAWSLogs processes a file containing ELB access logs.
+func (f *elbAccessLogUnmarshaler) UnmarshalAWSLogs(reader io.Reader) (plog.Logs, error) {
+	scanner := bufio.NewScanner(reader)
+
+	// Initialize scopeLogsByResource
+	scopeLogsByResource := map[string]plog.ScopeLogs{}
```

Comment:
All logs coming from the same should be part of the same resource. I don't think this map is needed